### PR TITLE
Add liger fused linear/crossentropy

### PR DESCRIPTION
**Pull request by apaz-cli**
Does not yet support z loss. See: https://github.com/linkedin/Liger-Kernel/issues/527

Also I made an `OptimizationsConfig`. Maybe we could organize the configs a little. Gradually.

### CODE REVIEW

File: src/zeroband/config.py [hunk 1]
```diff
@@ -93,6 +97,11 @@ class TrainConfig(BaseConfig):
     attn_fn: AttnFnType = "flex"
 
 
+class OptimizationsConfig(BaseConfig):
```
Comment (samsja) [inline src/zeroband/config.py]: let's not call it optimization because its got confusing with optimizer stuff.

File: src/zeroband/train.py [hunk 2]
```diff
@@ -320,11 +317,15 @@ def train(config: Config):
                     flatten_labels = rearrange(labels, "b seq -> (b seq)")
 
                 with record_function("Loss calculation"):
+                    if (config.optimizations.fused_linear_ce and config.optim.z_loss):
+                        raise ValueError("Liger kernel does not yet support fused linear CE and z loss. See https://github.com/linkedin/Liger-Kernel/issues/527")
```
Comment (samsja) [inline src/zeroband/train.py]:
```suggestion
                        raise NotImplementedError("Liger kernel does not yet support fused linear CE and z loss. See https://github.com/linkedin/Liger-Kernel/issues/527")
```
